compiler+runtime/src/cpp/jank/runtime/obj/ratio.cpp

```diff
@@ -0,0 +1,556 @@
+#include <jank/runtime/obj/ratio.hpp>
+#include <fmt/compile.h>
+#include <limits>
+
+namespace jank::runtime
+{
+
+  constexpr native_real EPSILON = std::numeric_limits<native_real>::epsilon();
+
+  obj::ratio_data::static_object(native_integer numerator, native_integer denominator)
+    : numerator{ numerator }
+    , denominator{ denominator }
+  {
+    if(denominator == 0)
+    {
+      throw std::invalid_argument("Ratio denominator cannot be zero.");
+    }
+    simplify();
+  }
+
+  object_ptr obj::ratio::create(native_integer const numerator, native_integer const denominator)
+  {
+    auto ratio_data = make_box<obj::ratio_data>(numerator, denominator);
+    if(ratio_data->denominator == 1)
+    {
+      return make_box<obj::integer>(ratio_data->numerator);
+    }
+    auto r(make_box<obj::ratio>());
```

Comment:
Why not use the ctor here which takes in the numerator/denominator?

Reply:
i'm trying to not include a ctor for `ratio` that takes num and denom, so that everything goes through `ratio_data` instead. that way we can avoid the possibility that someone accidentally init a ratio with 1 as denom. 

i added a ctor with a `ratio_data` as the only arg. does this work better?

Reply:
Yep, that can work!